src/test/java/pl/jakubkozlowski/learning/firststeps/mapper/ChampionMapperTest.java

```diff
@@ -19,88 +17,121 @@
     }
 
     @Test
     public void whenFindOne_thenReturnChampion() {
         //given
-        championMapper.create(expected);
+        championMapper.persist(expectedAatrox);
         //when
-        Champion actual = championMapper.findOne(1L);
+        Champion actual = championMapper.findById(id1);
         //then
         assertThat(actual)
-                .isEqualTo(expected);
+                .isEqualTo(expectedAatrox);
     }
 
     @Test
-    public void whenFindAll_thenReturnChampions(){
+    public void whenFindAll_thenReturnChampionList(){
         //given
-        expected2 = new Champion(2L, "Ahri");
-        championMapper.create(expected);
-        championMapper.create(expected2);
+        championMapper.persist(expectedAatrox);
+        championMapper.persist(expectedAhri);
         //when
         List<Champion> actual = championMapper.findAll();
         //then
+        assertThat(actual.size())
+                .isEqualTo(2);
         assertThat(actual.get(0))
-                .isEqualTo(expected);
+                .isEqualTo(expectedAatrox);
         assertThat(actual.get(1))
-                .isEqualTo(expected2);
+                .isEqualTo(expectedAhri);
+    }
+
+    @Test
+    public void whenPersist_thenIdAutoincrementsAndReturnsChampion(){
+        //when
+        championMapper.persist(expectedAhri);
+        //then
+        assertThat(championMapper.findByName(expectedAhri.getName()))
+                .isEqualTo(new Champion(id1, AHRI));
+
     }
 
     @Test(expected = DuplicateKeyException.class)
-    public void whenCreateChampionWithSameId_thenExceptionOccurs(){
-        //given
-        championMapper.create(expected);
-        expected2 = new Champion(1L, "Ahri");
+    public void whenPersistWithTheSameName_thenExceptionOccur(){
         //when
-        championMapper.create(expected2);
+        championMapper.persist(expectedAhri);
+        championMapper.persist(expectedAhri);
         //then
-        //DuplicateKeyException occurs
+
     }
 
     @Test
     public void whenUpdate_thenUpdateChampionEntity(){
         //given
-        championMapper.create(expected);
-        expected2 = new Champion(3L, "Ahri");
+        championMapper.persist(expectedAatrox);
         //when
-        championMapper.update(1L, expected2);
-        System.out.println(championMapper.findOne(1L));
-        Champion prev = championMapper.findOne(1L);
-        Champion actual = championMapper.findOne(3L);
+        championMapper.update(id1, expectedAhri);
+        Champion prev = championMapper.findById(id1);
+        Champion actual = championMapper.findById(id2);
         //then
-        assertThat(prev)
-                .isEqualTo(null);
+        assertThat(prev).isNull();
         assertThat(actual)
-                .isEqualTo(expected2);
+                .isEqualTo(expectedAhri);
 
     }
 
+    @Test(expected = DuplicateKeyException.class)
+    public void whenUpdateToUsedId_thenExceptionOccurs(){
```

Comment:
Redundant. Primary key should be immutable by deafult so there is no need for testing it for change